java-checks/src/main/java/org/sonar/java/checks/InnerStaticClassesCheck.java

```diff
@@ -69,12 +69,21 @@ public void visitClass(ClassTree tree) {
     scan(tree.members());
     Boolean oneReference = atLeastOneReference.pop();
     outerClasses.pop();
-    if (!symbol.isStatic() && !oneReference && !outerClasses.isEmpty() && isFirstParentStatic(outerClasses)) {
+    if (!symbol.isStatic() && !oneReference && !outerClasses.isEmpty() && isFirstParentStatic(outerClasses) && !hasNonStaticSuperClass(tree)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1067)